### ISSUE TITLE
Issues with validating a document with embedded documents

### ISSUE THREAD

**Issue by ragulka** [Automattic/mongoose#1883]
It looks like there are some serious issues with embedded document validation. Here are 3 "test cases":

```javascript
var mongoose = require("mongoose"),
    Schema = mongoose.Schema;

var CommentSchema = new Schema({
  title: { type: String, required: true },
  body: { type: String, required: true },
  status: String
});
var UserSchema = new Schema({
  name: { type: String, required: true },
  comments: [CommentSchema]
});
var PostSchema = new Schema({
  title: { type: String, required: true },
  contributors: [UserSchema],
  comments: [CommentSchema]
});

Comment = mongoose.model('Comment', CommentSchema);
User = mongoose.model('User', UserSchema);
Post = mongoose.model('Post', PostSchema);

var user = new User({ comments: [{ status: 'published' }, { status: 'published' }] });
user.validate(function(err) {
  console.log('1 ----------')
  console.log(err);
});

var post = new Post({
  contributors: [{}, {}],
  comments: [{ status: 'published' }, { status: 'published' }]
});
post.validate(function(err) {
  console.log('2 ----------')
  console.log(err);
});

var post = new Post({
  comments: [{ status: 'published' }, { status: 'published' }]
});
post.validate(function(err) {
  console.log('3 ----------')
  console.log(err);
});
```
And their respective results:
```javascript
1 ----------
{ message: 'Validation failed',
  name: 'ValidationError',
  errors: 
   { name: 
      { message: 'Path `name` is required.',
        name: 'ValidatorError',
        path: 'name',
        type: 'required',
        value: undefined },
     'comments.0.body': 
      { message: 'Path `body` is required.',
        name: 'ValidatorError',
        path: 'body',
        type: 'required',
        value: undefined },
     'comments.0.title': 
      { message: 'Path `title` is required.',
        name: 'ValidatorError',
        path: 'title',
        type: 'required',
        value: undefined } },
  key: 'undefined.0.undefined' }
2 ----------
{ message: 'Validation failed',
  name: 'ValidationError',
  errors: 
   { title: 
      { message: 'Path `title` is required.',
        name: 'ValidatorError',
        path: 'title',
        type: 'required',
        value: undefined },
     'comments.0.body': 
      { message: 'Path `body` is required.',
        name: 'ValidatorError',
        path: 'body',
        type: 'required',
        value: undefined },
     'comments.0.title': 
      { message: 'Path `title` is required.',
        name: 'ValidatorError',
        path: 'title',
        type: 'required',
        value: undefined },
     'comments.1.body': 
      { message: 'Path `body` is required.',
        name: 'ValidatorError',
        path: 'body',
        type: 'required',
        value: undefined },
     'comments.1.title': 
      { message: 'Path `title` is required.',
        name: 'ValidatorError',
        path: 'title',
        type: 'required',
        value: undefined },
     'contributors.0.name': 
      { message: 'Path `name` is required.',
        name: 'ValidatorError',
        path: 'name',
        type: 'required',
        value: undefined } },
  key: 'undefined.0.undefined.0.undefined' }
3 ----------
{ message: 'Validation failed',
  name: 'ValidationError',
  errors: 
   { title: 
      { message: 'Path `title` is required.',
        name: 'ValidatorError',
        path: 'title',
        type: 'required',
        value: undefined },
     'comments.0.body': 
      { message: 'Path `body` is required.',
        name: 'ValidatorError',
        path: 'body',
        type: 'required',
        value: undefined },
     'comments.0.title': 
      { message: 'Path `title` is required.',
        name: 'ValidatorError',
        path: 'title',
        type: 'required',
        value: undefined } },
  key: 'undefined.0.undefined' }
```

1. When there is only one embedded document defined in a schema (User), validation returns errors for the first embedded document that failed validation only. It should return errors for all embedded documents that failed validation

2. When there are multiple embedded documents defined in a schema (Post), AND there are embedded documents supplied for both of them (contributors AND comments) in the constructor, validation returns validation errors for all invalid embedded documents, like it should.

3. Same as above, but this time, embedded documents for only one subdocument array (comments) is supplied in the constructor. Validation returns the same result as in 1. It should work as in 2.

4. In all cases, a mysterious `key: 'undefined.0.undefined'` is added to the end of the returned error object.

5. If I use `save` instead of `validate` - all of the above "fail" - they only return the first validation error of the first embedded document. They don't even return the parent document's validation errors.

I'd gladly contribute actual tests as a PR, but I couldn't figure out where I should do it.

**Comment by abdullam** [comment]
anyone solved this issue?

**Comment by vkarpov15** [comment]
As far as I can tell, this issue has been fixed in 4.5.8, here's what the OP script prints:

```
$ node gh-1883.js 
1 ----------
{ ValidationError: User validation failed
    at MongooseError.ValidationError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validation.js:22:11)
    at model.Document.invalidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1416:32)
    at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1292:17
    at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:701:7)
    at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
    at Array.forEach (native)
    at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
    at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
    at Module.runMain (module.js:577:11)
    at run (node.js:348:7)
    at startup (node.js:140:9)
    at node.js:463:3
  message: 'User validation failed',
  name: 'ValidationError',
  errors: 
   { name: 
      { ValidatorError: Path `name` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `name` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'name',
        value: undefined },
     'comments.0.body': 
      { ValidatorError: Path `body` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `body` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'body',
        value: undefined },
     'comments.0.title': 
      { ValidatorError: Path `title` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `title` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'title',
        value: undefined },
     'comments.1.body': 
      { ValidatorError: Path `body` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `body` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'body',
        value: undefined },
     'comments.1.title': 
      { ValidatorError: Path `title` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `title` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'title',
        value: undefined } } }
2 ----------
{ ValidationError: Post validation failed
    at MongooseError.ValidationError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validation.js:22:11)
    at model.Document.invalidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1416:32)
    at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1292:17
    at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:701:7)
    at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
    at Array.forEach (native)
    at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
    at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
    at Module.runMain (module.js:577:11)
    at run (node.js:348:7)
    at startup (node.js:140:9)
    at node.js:463:3
  message: 'Post validation failed',
  name: 'ValidationError',
  errors: 
   { title: 
      { ValidatorError: Path `title` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `title` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'title',
        value: undefined },
     'comments.0.body': 
      { ValidatorError: Path `body` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `body` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'body',
        value: undefined },
     'comments.0.title': 
      { ValidatorError: Path `title` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `title` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'title',
        value: undefined },
     'comments.1.body': 
      { ValidatorError: Path `body` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `body` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'body',
        value: undefined },
     'comments.1.title': 
      { ValidatorError: Path `title` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `title` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'title',
        value: undefined },
     'contributors.0.name': 
      { ValidatorError: Path `name` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `name` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'name',
        value: undefined },
     'contributors.1.name': 
      { ValidatorError: Path `name` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `name` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'name',
        value: undefined } } }
3 ----------
{ ValidationError: Post validation failed
    at MongooseError.ValidationError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validation.js:22:11)
    at model.Document.invalidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1416:32)
    at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1292:17
    at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:701:7)
    at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
    at Array.forEach (native)
    at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
    at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
    at Module.runMain (module.js:577:11)
    at run (node.js:348:7)
    at startup (node.js:140:9)
    at node.js:463:3
  message: 'Post validation failed',
  name: 'ValidationError',
  errors: 
   { title: 
      { ValidatorError: Path `title` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `title` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'title',
        value: undefined },
     'comments.0.body': 
      { ValidatorError: Path `body` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `body` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'body',
        value: undefined },
     'comments.0.title': 
      { ValidatorError: Path `title` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `title` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'title',
        value: undefined },
     'comments.1.body': 
      { ValidatorError: Path `body` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `body` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'body',
        value: undefined },
     'comments.1.title': 
      { ValidatorError: Path `title` is required.
          at MongooseError.ValidatorError (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/error/validator.js:24:11)
          at validate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:700:13)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:738:9
          at Array.forEach (native)
          at SchemaString.SchemaType.doValidate (/home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/schematype.js:706:19)
          at /home/val/Workspace/10gen/troubleshoot-mongoose/node_modules/mongoose/lib/document.js:1290:9
          at _combinedTickCallback (internal/process/next_tick.js:67:7)
          at process._tickCallback (internal/process/next_tick.js:98:9)
          at Module.runMain (module.js:577:11)
          at run (node.js:348:7)
          at startup (node.js:140:9)
          at node.js:463:3
        message: 'Path `title` is required.',
        name: 'ValidatorError',
        properties: [Object],
        kind: 'required',
        path: 'title',
        value: undefined } } }
```
